lib/split/cache.rb

```diff
@@ -18,5 +18,9 @@ def self.fetch(namespace, key)
 
       @cache[namespace][key] = yield
     end
+
+    def self.clear_key(namespace, key)
+      @cache[namespace]&.delete(key)
+    end
```

Comment:
Hmm ... I'm not sure about this.  It will clear the key in the local server, but other servers in the cluster will still have stale cached data.  Think it may be better for all servers to be in the same state, synchronized by restart, rather than having all stale, and one reset.

Reply:
I guess, in a low-traffic cluster, some set of servers could cache before a change and others after, so this isn't really a new problem related to this logic.  More of a general local-cache issue.

Reply:
We talked about this off line --- all servers will update to the latest value in a couple of requests at most.

Reply:
So, I'd like to avoid clearing specific keys.  I think we just `Split::Cache.clear` anytime stuff gets stale, rather than go down the rabbit hole of keeping track of which keys when.  Will be a little bit less efficient, but way less error prone.

Reply:
To be explicit, let's get rid of this method, and just invalidate the whole cache anywhere it's needed.

Reply:
Is there any specific reason? Within the experiment class, it's not ambiguous which key is being cleared.

Reply:
Just to expand on my reasoning for choosing this way --- Cache is a standard concept in computing and there is a certain expected behavior from them. They all have an eviction policy, and it is defined as criterion (key) --> evict(key). In this case we have a explicit staleness criteria. Unless there is a specific usage pattern that is different (which I don't see here), why should we deviate from the norm?

Reply:
Ah, mr primary concern with the original implementation/usage is that we'd need maintain the clearing logic, as well.  So if we cached another experiment value, we'd also need to know to add another line to clear it from the cache in the reset logic.  

Looks like your new approach resolves that method, as long as we use the same name for the experiment everywhere.  I'm cool with this new strategy.